fix(docs): show where `getNodesByType` comes from (#25344)
gatsbyjs/gatsby: docs/docs/creating-a-source-plugin.md

> The code to support this behavior looks like this:

Reviewer: ```suggestion
Here's some pseudo code that shows this behavior:
```

Author: I've committed this change. Thanks for the suggestion!
